### PR TITLE
Don't make a ghost indicator if another vehicle is on the run

### PR DESCRIPTION
**Pull request by arkadyan**
Asana ticket: [🐞 Showing both a ghost and a regular bus on the route](https://app.asana.com/0/1148853526253426/1161461620801441)

### CODE REVIEW

File: assets/src/components/routeLadder.tsx [hunk 1]
```diff
@@ -177,6 +177,27 @@ export const groupByPosition = (
...
+
+  const otherVehicles = onlyVehicles(vehiclesAndGhosts).filter(
+    ({ id }) => id !== vehicle.id
+  )
+  const otherRunIds = runIds(otherVehicles)
+
+  return !otherRunIds.includes(vehicle.runId)
+}
+
+const onlyVehicles = (vehiclesAndGhosts: VehicleOrGhost[]): Vehicle[] =>
```
Comment (skyqrose) [inline assets/src/components/routeLadder.tsx]: Is this needed? Ghosts wouldn't be made with a matching run, and even if there was one, you probably wouldn't want to draw the virtual ghost.

File: assets/src/components/routeLadder.tsx [hunk 2]
```diff
@@ -177,6 +177,27 @@ export const groupByPosition = (
   }
 }
 
+const runNotSharedByAnotherVehicle = (vehiclesAndGhosts: VehicleOrGhost[]) => (
+  vehicle: Vehicle
+): boolean => {
+  if (vehicle.runId === null) {
```
Comment (skyqrose) [inline assets/src/components/routeLadder.tsx]: When do we ever have a null run id?

Comment (arkadyan) [reply assets/src/components/routeLadder.tsx]: I don't have a great sense, but it seems like they come in occasionally, but this is also just about dealing with the type definition.

File: assets/src/components/routeLadder.tsx (hunk 1 above)
Comment (arkadyan) [reply assets/src/components/routeLadder.tsx]: Yeah, I guess that's true.

**Pull request merged**